### ISSUE TITLE
Can't install coverage v4.3 into under python 3.4.3

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#540]
*Originally reported by* **Cédric RICARD (Bitbucket: [ricard](https://bitbucket.org/ricard), GitHub: [ricard](https://github.com/ricard))**

----------------------------------------

My virtualenv failled to build with coverage 4.3 (was successful with 4.2) :

```
Collecting coverage==4.3
  Using cached coverage-4.3.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-5oknbhqx/coverage/setup.py", line 50, in <module>
        paras = contributors.read().split("\n\n")
      File "/var/lib/jenkins/jobs/akema_gestion/workspace/tmpenv/lib/python3.4/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xc4 in position 846: ordinal not in range(128)

```
 

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/540

**Comment by nedbat** [comment]
I can't reproduce this on my Mac.  Can you provide more information about your OS and environment variables?

**Comment by nedbat** [comment]
*Original comment by* **Chen Huang (Bitbucket: [chinux23](https://bitbucket.org/chinux23), GitHub: [chinux23](https://github.com/chinux23))**

----------------------------------------

happens on linux for me as well. Works on Mac.

```
$ pip3 install coverage
Collecting coverage
  Downloading coverage-4.3.tar.gz (371kB)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-xu2tywse/coverage/setup.py", line 50, in <module>
        paras = contributors.read().split("\n\n")
      File "/usr/lib/python3.5/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xc4 in position 846: ordinal not in range(128)
```

OS info. Docker image: ubuntu:latest
```
Running with gitlab-ci-multi-runner 1.9.0 (82714ae)
Using Docker executor with image ubuntu:latest ...
Pulling docker image ubuntu:latest ...
Running on runner-4e4528ca-project-2212302-concurrent-0 via runner-4e4528ca-machine-1482878059-e8e6e297-digital-ocean-4gb...
```

Setups after image installed.
```
  6     - apt-get update
  7     - apt-get install -y --no-install-recommends build-essential cmake git wget
  8     - apt-get install -y --no-install-recommends python3-dev python3-pip
  9     - pip3 install --upgrade pip
 10     - pip3 install setuptools
 11     - pip3 install coverage
```

**Comment by nedbat** [comment]
*Original comment by* **Cédric RICARD (Bitbucket: [ricard](https://bitbucket.org/ricard), GitHub: [ricard](https://github.com/ricard))**

----------------------------------------

I'm running Jenkins on Linux/Debian :

```
Linux jenkins 2.6.32-39-pve #1 SMP Fri May 8 11:27:35 CEST 2015 x86_64 GNU/Linux
```

**Comment by nedbat** [comment]
If you are having this problem, please show your environment variables.  What is the output of "env"?

**Comment by nedbat** [comment]
*Original comment by* **Cédric RICARD (Bitbucket: [ricard](https://bitbucket.org/ricard), GitHub: [ricard](https://github.com/ricard))**

----------------------------------------

```
jenkins@jenkins:~$ env
COMP_WORDBREAKS=
"'><;|&(:
SHELL=/bin/bash
TERM=xterm-256color
DERBY_HOME=/usr/lib/jvm/java-7-oracle/db
LC_ALL=C
USER=jenkins
MAIL=/var/mail/jenkins
PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/usr/lib/jvm/java-7-oracle/bin:/usr/lib/jvm/java-7-oracle/db/bin:/usr/lib/jvm/java-7-oracle/jre/bin
PWD=/var/lib/jenkins
JAVA_HOME=/usr/lib/jvm/java-7-oracle
LANG=en_US.UTF-8
SHLVL=1
HOME=/var/lib/jenkins
LANGUAGE=en_US.UTF-8
LOGNAME=jenkins
J2SDKDIR=/usr/lib/jvm/java-7-oracle
LC_CTYPE=C
J2REDIR=/usr/lib/jvm/java-7-oracle/jre
_=/usr/bin/env
```

**Comment by nedbat** [comment]
*Original comment by* **Chen Huang (Bitbucket: [chinux23](https://bitbucket.org/chinux23), GitHub: [chinux23](https://github.com/chinux23))**

----------------------------------------

```
$ env
CI_PROJECT_NAME=dovahkiin
CI_REGISTRY=registry.gitlab.com
CI_BUILD_TOKEN=xxxxxxxxxxxxxxxxxxxx
HOSTNAME=runner-4e4528ca-project-2212302-concurrent-0
CI_PROJECT_URL=https://gitlab.com/chinux23/dovahkiin
CI_BUILD_BEFORE_SHA=c2a3700b4fa75b89b92c80eca6c1cd23f15af1e0
CI_SERVER_VERSION=8.15.1-ee
CI_BUILD_ID=7917272
OLDPWD=/
GITLAB_USER_EMAIL=chinux@gmail.com
CI_PROJECT_ID=2212302
CI_RUNNER_ID=40788
CI_PIPELINE_ID=5563181
CI_BUILD_REF_NAME=master
CI_BUILD_REF=c2a3700b4fa75b89b92c80eca6c1cd23f15af1e0
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
CI_BUILD_STAGE=test
CI_PROJECT_DIR=/builds/chinux23/dovahkiin
CI_RUNNER_TAGS=git-annex, mongo, postgres, mysql, ruby, linux, docker, shared
CI_REGISTRY_IMAGE=registry.gitlab.com/chinux23/dovahkiin
PWD=/builds/chinux23/dovahkiin
CI_DEBUG_TRACE=false
CI_SERVER_NAME=GitLab
CI_PROJECT_PATH=chinux23/dovahkiin
GITLAB_CI=true
CI_SERVER_REVISION=3e62eee
SHLVL=1
CI_BUILD_NAME=YiBei
HOME=/root
CI_SERVER=yes
CI=true
CI_PROJECT_NAMESPACE=chinux23
CI_BUILD_REF_SLUG=master
CI_BUILD_REPO=https://gitlab-ci-token:xxxxxxxxxxxxxxxxxxxx@gitlab.com/chinux23/dovahkiin.git
CI_RUNNER_DESCRIPTION=shared-runners-manager-2.gitlab.com
GITLAB_USER_ID=5020
_=/usr/bin/env
```

**Comment by nedbat** [comment]
This is fixed in [502d5e2aad49 (bb)](https://bitbucket.org/ned/coveragepy/commits/502d5e2aad49).  If you could try installing from the tip of bitbucket, I would really appreciate it.

**Comment by nedbat** [comment]
*Original comment by* **gouthamr (Bitbucket: [gouthamr](https://bitbucket.org/gouthamr), GitHub: [gouthamr](https://github.com/gouthamr))**

----------------------------------------

This issue affects certain OpenStack projects :

Our CI jobs run on Ubuntu Xenial, with py35

Sample: https://review.openstack.org/#/c/415138/3
Logs: http://logs.openstack.org/38/415138/3/check/gate-python-manilaclient-python35/15278f0/console.html#_2016-12-28_03_04_42_940004

**Comment by nedbat** [comment]
*Original comment by* **gouthamr (Bitbucket: [gouthamr](https://bitbucket.org/gouthamr), GitHub: [gouthamr](https://github.com/gouthamr))**

----------------------------------------

Ned, any chance you're releasing your fix to PyPi soon?

**Comment by nedbat** [comment]
*Original comment by* **Cédric RICARD (Bitbucket: [ricard](https://bitbucket.org/ricard), GitHub: [ricard](https://github.com/ricard))**

----------------------------------------

The commit [502d5e2aad49 (bb)](https://bitbucket.org/ned/coveragepy/commits/502d5e2aad49) works for me. Thank you.

**Comment by nedbat** [comment]
Issue #542 was marked as a duplicate of this issue.

**Comment by nedbat** [comment]
This fix has been released as part of version 4.3.1: https://pypi.python.org/pypi/coverage/4.3.1
